Detect windows using `os.name` rather than `line.separator`.

nedtwigg opened the pull request:
Added `FileSignature.machineIsWin`, which detects the OS using `os.name`, and replaces the now-deprecated `LineEnding.nativeIsWin`, which detected the OS using `line.separator`.

J-N-K commented on lib/src/main/java/com/diffplug/spotless/FileSignature.java:
```diff
@@ -106,14 +107,21 @@ public File getOnlyFile() {
 	/** Transforms a native path to a unix one. */
 	public static String pathNativeToUnix(String pathNative) {
-		return LineEnding.nativeIsWin() ? pathNative.replace('\\', '/') : pathNative;
+		return machineIsWin() ? pathNative.replace('\\', '/') : pathNative;
```
```suggestion
		return pathNative.replace(File.separatorChar, '/');
```

J-N-K commented on lib/src/main/java/com/diffplug/spotless/FileSignature.java:
```diff
@@ -106,14 +107,21 @@ public File getOnlyFile() {
 	/** Transforms a unix path to a native one. */
 	public static String pathUnixToNative(String pathUnix) {
-		return LineEnding.nativeIsWin() ? pathUnix.replace('/', '\\') : pathUnix;
+		return machineIsWin() ? pathUnix.replace('/', '\\') : pathUnix;
```
```suggestion
		return pathUnix.replace('/', File.separatorChar);
```

nedtwigg commented:
Ooh, I like the `File.separatorChar`, thanks.  We'll have a bugfix release out by Monday, there's one other bug I need to look into.

nedtwigg commented:
There was a related bug, which was completely clobbering the git-native line endings anyway.  Just fixed in previous commit.  With these fixes combined, I think we have fixed the problem completely.

nedtwigg commented:
Released in `plugin-maven 2.0.1` and `plugin-gradle 4.5.1`
